modules/host/update.go

```diff
@@ -65,37 +89,15 @@ func (h *Host) ReceiveConsensusSetUpdate(cc modules.ConsensusChange) {
 
 	// Check the applied blocks and see if any of the contracts we have are
 	// ready for storage proofs.
-	shouldSave := false
 	for _ = range cc.AppliedBlocks {
 		h.blockHeight++
-
 		for _, obligation := range h.obligationsByHeight[h.blockHeight] {
-			// Submit a storage proof for the obligation.
-			err := h.createStorageProof(obligation, h.blockHeight)
-			if err != nil {
-				fmt.Println(err)
-				return
-			}
-
-			// Storage proof was successful, so increment profit tracking
-			h.profit.Add(obligation.FileContract.Payout)
-
-			// Delete the obligation.
-			fullpath := filepath.Join(h.saveDir, obligation.Path)
-			stat, err := os.Stat(fullpath)
-			if err != nil {
-				fmt.Println(err)
-				return
-			}
-			h.deallocate(uint64(stat.Size()), obligation.Path) // TODO: file might actually be the wrong size.
-			shouldSave = true
-			delete(h.obligationsByID, obligation.ID)
+			go h.threadedCreateStorageProof(obligation, h.blockHeight)
```

Comment:
since `threadedDeleteObligation` is a separate function now, you could write:
```go
go func() {
    err := h.threadedCreateStorageProof(obligation, h.blockHeight)
    if err != nil {
        fmt.Println(err)
        return
    }
    h.threadedDeleteObligation(obligation)
}()
```
then `threadedCreateStorageProof` could just return an error instead of printing all over the place